assembl/static2/js/app/components/debate/brightMirror/sideComment/sideCommentBox.jsx

```diff
@@ -195,6 +205,35 @@ class DumbSideCommentBox extends React.Component<Props, State> {
       });
   };
 
+  submitReply = (): void => {
+    const { ideaId, contentLocale, createPost, refetchPost } = this.props;
+    const { body, extractIndex } = this.state;
+    const currentExtract = this.getCurrentExtract(extractIndex);
+    const currentComment = this.getCurrentComment(currentExtract);
+
+    const bodyIsEmpty = editorStateIsEmpty(body);
+    if (bodyIsEmpty) {
+      displayAlert('warning', I18n.t('debate.thread.fillBody'));
+      return;
+    }
+
+    const postVars = {
+      ideaId: ideaId,
+      extractId: currentExtract.id,
+      parentId: currentComment.id,
+      body: convertContentStateToHTML(body.getCurrentContent()),
+      contentLocale: contentLocale
+    };
+    createPost({ variables: postVars }).then(() => {
+      this.setState({
+        replying: false
+      });
+      window.getSelection().removeAllRanges();
```

Comment:
maybe put whatever comes after this ```this.setState```in the setState callback function ? I think it's better. @amensouissi can you confirm?

Reply:
I believe it's about the timing, if you need those to run only once the state is updated